chirp/drivers/hf90.py

```diff
@@ -0,0 +1,567 @@
+            mem.duplex = "+"
+            mem.offset = (int(_tx) - int(_rx)) * 100
+        elif _tx < _rx:
+            mem.duplex = "-"
+            mem.offset = (int(_rx) - int(_tx)) * 100
+        else:
+            mem.duplex = ""
+            mem.offset = 0
+
+        if _mem.usb:
+            mem.mode = "USB"
+        else:
+            mem.mode = "LSB"
+
+        n = _mem.vnum & 0xff
+        if n == 0:
+            n = number
+
+        rsg = RadioSettingGroup("Extra", "extra")
+        rs = RadioSetting("vnum", "Number",
+                          RadioSettingValueInteger(self._lower,
+                                                   self._upper, n))
+        rsg.append(rs)
+        rs = RadioSetting("selcall", "Selcall",
+                          RadioSettingValueBoolean(_mem.selcall))
+        rsg.append(rs)
+        rs = RadioSetting("scan", "Scan",
+                          RadioSettingValueBoolean(_mem.scan))
+        rsg.append(rs)
+        mem.extra = rsg
+
+        return mem
+
+    def _get_tx_freq(self, mem):
+        if mem.duplex == "":
+            return mem.freq
+        elif mem.duplex == "split":
+            return mem.offset
+        elif mem.duplex == "+":
+            return mem.freq + mem.offset
+        elif mem.duplex == "-":
+            return mem.freq - mem.offset
+        return 0
+
+    def set_memory(self, mem):
+        _mem = self._memobj.memory[mem.number - 1]
+
+        _mem.set_raw("\x00" * 8)
+
+        if mem.empty:
+            return
+
+        rx = int(mem.freq / 100)
+        tx = int(self._get_tx_freq(mem) / 100)
+
+        _mem.tx_equals_rx = tx == rx
+
+        _mem.rx0_7 = (rx >> 0) & 0xff
+        _mem.rx8_15 = (rx >> 8) & 0xff
+        _mem.rx16_19 = (rx >> 16) & 0xf
+
+        _mem.tx0_3 = (tx >> 0) & 0xf
+        _mem.tx4_7 = (tx >> 4) & 0xf
+        _mem.tx8_11 = (tx >> 8) & 0xf
+        _mem.tx12_15 = (tx >> 12) & 0xf
+        _mem.tx16_19 = (tx >> 16) & 0xf
+
+        _mem.usb = mem.mode == "USB"
+
+        _mem.vnum = mem.number
+
+        for setting in mem.extra:
+            setattr(_mem, setting.get_name(), int(setting.value))
+
+    def validate_memory(self, mem):
+        msgs = chirp_common.CloneModeRadio.validate_memory(self, mem)
+
+        if 0 < int(self._get_tx_freq(mem) / 100) < self._min_tx_freq:
+            msgs.append(chirp_common.ValidationError(
+                "tx frequency too low"))
+
+        if mem.extra and isinstance(self, EarlyHF90Radio):
+            if self._memobj.settings.selcall_id == 0x0fff:
+                if mem.extra["scan"].value or mem.extra["selcall"].value:
+                    msgs.append(chirp_common.ValidationError(
+                        "standard type but scan or selcall enabled"))
+            else:
+                if mem.extra["scan"].value and not mem.extra["selcall"].value:
+                    msgs.append(chirp_common.ValidationError(
+                        "scan enabled but selcall disabled"))
+
+        return msgs
+
```

Comment:
The unit failure is related to detection of another radio's image as your driver, because it is the same as your declared `_memsize`. Long ago, we detected models by their memory size, and so that's what the default implementation still does for compatibility. We now append a metadata blob at the end of the file, so radios added after should really opt out of file size detection. So you should add this here:
```
@classmethod
def match_model(cls, filedata, filename):
    return False
```
So that your driver only gets detected by the metadata and never by the size or content of the (memory part of the) file. That should clear up that unit fail.